ceph/ceph_admin/apply_spec.py

```diff
@@ -0,0 +1,74 @@
+"""
+Represent the ceph orch CLI action 'apply' using spec file.
+
+Module to deploy ceph service(s) using orchestration command
+"ceph orch apply -i spec.yaml"
+
+"""
+import logging
+import tempfile
+from typing import Dict
+
+from .common import config_dict_to_string
+from .helper import PrepareSpec
+from .typing_ import ServiceProtocol
+
+LOG = logging.getLogger()
+
+
+class ApplySpec:
+    """Add apply command support to provision service using spec file."""
+
+    def apply_spec(self: ServiceProtocol, config: Dict) -> None:
+        """
+        Execute the apply_spec method using the object's service name and provided input.
+
+        Args:
+            config:     Key/value pairs passed from the test suite.
+                        base_cmd_args   - key/value pairs to set for base command
+                        specs           - service specifications.
+
+        Example:
+            config:
+                command: apply_spec
+                service: orch
+                base_cmd_args:          # arguments to ceph orch
+                    concise: true
+                    verbose: true
+                specs:
+                 - service_type: host
+                   attach_ip_address: true
+                   labels: apply-all-labels
+                   nodes:
+                     - node2
+                     - node3
+        """
+        base_cmd = ["ceph", "orch"]
+
+        if config.get("base_cmd_args"):
+            base_cmd_args_str = config_dict_to_string(config.get("base_cmd_args"))
+            base_cmd.append(base_cmd_args_str)
+
+        base_cmd.append("apply -i")
+        specs = config["specs"]
```

Comment:
Instead of a dict, let's ensure the child class passes a file to this method. This way we can ensure the processing would happen in the child class. This would allow greater flexibility to them. The dependencies like which Jinja2 template to be used and the supported pre-processing can implemented easily in the child's method.

Reply:
As we discussed, apply_spec is part Orch.